src/components/TracePage/TraceTimelineViewer/VirtualizedTraceView.js

```diff
@@ -0,0 +1,367 @@
+// @flow
+
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as React from 'react';
+import cx from 'classnames';
+import { connect } from 'react-redux';
+
+import ListView from './ListView';
+import SpanBarRow from './SpanBarRow';
+import DetailState from './SpanDetail/DetailState';
+import SpanDetailRow from './SpanDetailRow';
+import Ticks from './Ticks';
+import TimelineRow from './TimelineRow';
+import type { XformedSpan, XformedTrace } from './transforms';
+import {
+  findServerChildSpan,
+  formatDuration,
+  getViewedBounds,
+  isErrorSpan,
+  spanContainsErredSpan,
+} from './utils';
+import type { Log } from '../../../types';
+import colorGenerator from '../../../utils/color-generator';
+
+import './VirtualizedTraceView.css';
+
+type RowState = {
+  isDetail: boolean,
+  span: XformedSpan,
+  spanIndex: number,
+};
+
+type VirtualizedTraceViewProps = {
+  childrenHiddenIDs: Set<string>,
+  childrenToggle: string => void,
+  detailLogItemToggle: (string, Log) => void,
+  detailLogsToggle: string => void,
+  detailProcessToggle: string => void,
+  detailStates: Map<string, ?DetailState>,
+  detailTagsToggle: string => void,
+  detailToggle: string => void,
+  findMatchesIDs: Set<string>,
+  ticks: number[],
+  trace?: XformedTrace,
```

Comment:
should this be required?

Reply:
No, if the trace needs to be loaded it is `undefined` until it is loaded.